### PR TITLE
vmc: New Package

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
"""The Virtual Monte Carlo (VMC) library"""

### CODE REVIEW

File: var/spack/repos/builtin/packages/vmc/package.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+from spack import *
+
+
+class Vmc(CMakePackage):
+    """The Virtual Monte Carlo (VMC) library"""
+
+    homepage = "https://github.com/vmc-project/vmc"
+    git      = 'https://github.com/vmc-project/vmc.git'
+    url      = "https://github.com/vmc-project/vmc/archive/v1-0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vmc/package.py]:
```suggestion
    url      = "https://github.com/vmc-project/vmc/archive/v1-0-p2.tar.gz"
```
This doesn't affect fetching, but it does affect the statistics reported by `spack url summary`

File: var/spack/repos/builtin/packages/vmc/package.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    patch('dict_fixes_101.patch', when='@1-0-p1')
+
+    depends_on('root@6.18.04: ~vmc')
+
+    def common_env_setup(self, env):
+        # So that root finds the shared library / rootmap
+        env.prepend_path("LD_LIBRARY_PATH", self.prefix.lib)
+
+    def setup_run_environment(self, env):
+        self.common_env_setup(env)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vmc/package.py]: Spack already adds this by default, this shouldn't be needed.

File: var/spack/repos/builtin/packages/vmc/package.py (hunk 1 above)
Comment (ChristianTackeGSI) [reply var/spack/repos/builtin/packages/vmc/package.py]: Fixed.

File: var/spack/repos/builtin/packages/vmc/package.py (hunk 2 above)
Comment (ChristianTackeGSI) [reply var/spack/repos/builtin/packages/vmc/package.py]: It was needed in older spack versions. I have used this recipe already for a while, just now upstreaming it.

Removed. Let's see.

(Note: I think, some parts need improvement: After removing all the `setup*` a `spack load vmc; spack unload vmc` still leaves some stuff in `$LD_LIBRARY_PATH`.)

**Review by adamjstewart — APPROVED** [review]

**Comment by ChristianTackeGSI** [comment]
Thanks! :-)

**Pull request merged**